### ISSUE TITLE
Cross-Site Scripting (XSS) via JavaScript URI in anchor tags

### ISSUE THREAD

**Issue by smaury** [nhn/tui.editor#1443]
## Describe the bug

TUI editor is vulnerable to Cross-Site Scripting (XSS) as it fails to properly sanitize the user-supplied input before including it in the HTML context.
More precisely, the application fails to verify the href attribute of anchor tags.

The vulnerable snippet of code follows:
```js
const XSS_ATTR_RX = /href|src|background/gi;
const XSS_VALUE_RX = /((java|vb|live)script|x):/gi;

[...]

function isXSSAttribute(attrName, attrValue) {
  return attrName.match(XSS_ATTR_RX) && attrValue.match(XSS_VALUE_RX);
}
```
The code which is supposed to avoid anchor tags having javascript:code() URIs does not consider that such URIs might be written with TABs (0x09) in between (i.e. <a href='j	avascript:alert(1)'>click me</a>)

## To Reproduce
1. Visit the following URL: https://nhn.github.io/tui.editor/latest/tutorial-example01-editor-basic
2. Input in the left pane the following string by making sure the character after the 'j' and before the 'a' is a TAB (0x09) and not a sequence of spaces (0x20): `<a href='j	avascript:alert(document.domain)'>click me</a>`
3. Click on the "click me" text in the right page
4. Notice the JavaScript code `alert(document.domain)` is executed

## Expected behavior
Injected JavaScript code should be executed.

## Screenshots
![image](https://user-images.githubusercontent.com/1758689/111462153-5f259800-871e-11eb-9d89-6fd87e439320.png)

## Desktop (please complete the following information):
 - OS: Windows, Linux, MacOS
 - Browser: Chrome 89.0.4389.90, Safari 14.0.3, Firefox 86.0.1

**Comment by js87zz** [comment]
@smaury 
It should be fixed. Thank you for reporting.

**Comment by smaury** [comment]
@js87zz I don't see any commit in last 7 days which might have fixed this. Can you point me in right direction?

**Comment by js87zz** [comment]
@smaury 
Sorry for late replying. Nothing has been worked on yet. If urgent, it can be used as a replacement with the `customHTML Sanitizer` option like below.

```js
import DOMPurify from 'dompurify';

const purifyOptions = { ... };
const editor = new Editor({
  // ...
  customHTMLSanitizer: html => {
    return DOMPurify.sanitize(html, purifyOptions) || ''
  }
});
```

**Comment by ready-research** [comment]
@js87zz We have opened a PR that will fix this security issue. Please have a look at it. If you are fine with it please comment @huntr-helper - LGTM , which will automatically open a PR in this repository.
If you need any changes please comment on this PR. Thanks.
Ref: https://github.com/418sec/tui.editor/pull/1

**Comment by ready-research** [comment]
@js87zz Can you please validate this issue in https://www.huntr.dev/bounties/1-other-tui.editor/

**Comment by js87zz** [comment]
@smaury 
I'm sorry for the late response. This issue was recently fixed in `v3.0.2`!
